zerocopy-derive/src/lib.rs

```diff
@@ -702,15 +702,20 @@ fn derive_split_at_inner(
         return Err(Error::new(Span::call_site(), "must not have #[repr(packed)] attribute"));
     }
 
+    if !(repr.is_c() || repr.is_transparent()) {
+        return Err(Error::new(Span::call_site(), "must have #[repr(C)] or #[repr(transparent)] in order to guarantee this type's layout is splitable"));
+    }
+
     let fields = ast.data.fields();
     let trailing_field = if let Some(((_, _, trailing_field), _)) = fields.split_last() {
         trailing_field
     } else {
         return Err(Error::new(Span::call_site(), "must at least one field"));
     };
 
-    // SAFETY: `#ty`, per the above check, is not packed; its trailing slice
-    // field is guaranteed to be well-aligned for its type.
+    // SAFETY: `#ty`, per the above checks, is `repr(C)` or `repr(transparent)`
+    // and is not packed; its trailing slice field is guaranteed to be
+    // well-aligned for its type.
```

Comment:
Does this need to also argue that the trailing field is `SplitAt`, which guarantees the alignment property recursively?

Reply:
https://github.com/google/zerocopy/compare/fd5f6b551b68939c61ea1d9bbcdc802d7d4823cb..9562f5e1784e47002f2e20edbdc6cfd7cb753afe